src/octoprint/util/__init__.py

```diff
@@ -510,6 +516,8 @@ def to_str(s_or_u, encoding="utf-8", errors="strict"):
 
 def to_unicode(s_or_u, encoding="utf-8", errors="strict"):
 	"""Make sure ``s_or_u`` is a unicode string."""
+	if PY3:
```

Comment:
Backwards? See above

Reply:
Giving a better look at this : 
Strings objects are always unicode in Py3. Then, unicode object become string object in Py3.
Bytes objects are just containers commonly used when playing with io
The pep describe that well
https://www.python.org/dev/peps/pep-0358/

With that in mind, to_unicode() calls will become useless in python3, and keeping the calls in the code could be a easy way to get rid of this mess when dropping Py2

Reply:
The thing is that `to_unicode` and `to_str` should probably have been called `to_unicode_string` and `to_byte_representation`. 

`to_unicode` should ensure that whatever it gets, py3 str or py3 bytes or py2 unicode or py2 str, a unicode string (py3 str & py2 unicode) is the result.

`to_str` should ensure that whatever it gets, py3 str or py3 bytes or py2 unicode or py2 str, a byte representation (py3 bytes & py2 str) is the result.